src/aggregator/aggregator/generic_elem.go

```diff
@@ -674,15 +745,20 @@ func (e *GenericElem) processValueWithAggregationLock(
 				prev := transformation.Datapoint{
 					Value: nan,
 				}
-				// lazily construct consumedValues since they are only needed by binary transforms.
-				if e.consumedValues == nil {
-					e.consumedValues = make(valuesByTime)
-				}
-				if _, ok := e.consumedValues[prevTimeNanos]; ok {
-					prev = e.consumedValues[prevTimeNanos][aggTypeIdx]
+				if flushState.prevStartTime > 0 {
+					prevFlushState, ok := e.flushState[flushState.prevStartTime]
+					if !ok {
+						ts := flushState.prevStartTime.ToTime()
+						instrument.EmitAndLogInvariantViolation(e.opts.InstrumentOptions(), func(l *zap.Logger) {
```

Comment:
I think this is another case where not panicking here (e.g. in prod) will result in a panic below because `prevFlushState` will have an empty `consumedValues`

Reply:
For something like this or some other bug it might be worth still keeping around that check against a nil `consumedValues` - just to prevent a panic. And we could just do `EmitAndLogInvariantViolation` if nil

Reply:
yea good catch